libphobos/configure.ac

```diff
@@ -53,291 +51,64 @@ AM_INIT_AUTOMAKE([1.11.1 foreign no-dist no-define no-dependencies -Wall -Wno-po
-  AC_EGREP_CPP(FOUND_LIBBACKTRACE_RESULT_GDC,
-  [
-  #include <backtrace-supported.h>
-  #if BACKTRACE_USES_MALLOC
-    FOUND_LIBBACKTRACE_RESULT_GDC
-  #endif
-  ], BACKTRACE_USES_MALLOC=true, BACKTRACE_USES_MALLOC=false)
-  AC_MSG_RESULT($BACKTRACE_USES_MALLOC)
-
-  AC_MSG_CHECKING([libbacktrace: BACKTRACE_SUPPORTS_THREADS])
-  AC_EGREP_CPP(FOUND_LIBBACKTRACE_RESULT_GDC,
-  [
-  #include <backtrace-supported.h>
-  #if BACKTRACE_SUPPORTS_THREADS
-    FOUND_LIBBACKTRACE_RESULT_GDC
-  #endif
-  ], BACKTRACE_SUPPORTS_THREADS=true, BACKTRACE_SUPPORTS_THREADS=false)
-  AC_MSG_RESULT($BACKTRACE_SUPPORTS_THREADS)
-fi
-
-AM_CONDITIONAL([BACKTRACE_SUPPORTED], [$BACKTRACE_SUPPORTED])
-
-if $BACKTRACE_SUPPORTED; then
-  if test "x$with_libbacktrace" != xno; then
-    LIBBACKTRACE_LIB="$with_libbacktrace/.libs/libbacktrace.a"
-  else
-    LIBBACKTRACE_LIB="../../libbacktrace/.libs/libbacktrace.a"
-  fi
-else
-  LIBBACKTRACE_LIB=""
-fi
-
-AC_SUBST(LIBBACKTRACE_LIB)
-AC_SUBST(BACKTRACE_SUPPORTED)
-AC_SUBST(BACKTRACE_USES_MALLOC)
-AC_SUBST(BACKTRACE_SUPPORTS_THREADS)
-
-dnl Thread model.
-AC_MSG_CHECKING([for thread model used by GDC])
-d_thread_model=`$GDC -v 2>&1 | sed -n 's/^Thread model: //p'`
-AC_MSG_RESULT([$d_thread_model])
-
-# Map from thread model to thread interface.
-DRUNTIME_CONFIGURE_THREADS([$d_thread_model])
-
-D_EXTRA_OBJS=
-AC_SUBST(D_EXTRA_OBJS)
-DRUNTIME_OBJS=
-AC_SUBST(DRUNTIME_OBJS)
-
-AC_MSG_CHECKING([for ARM unwinder])
-AC_TRY_COMPILE([#include <unwind.h>],[
-#if __ARM_EABI_UNWINDER__
-#error Yes, it is.
-#endif
-],
-  [AC_MSG_RESULT([no])
-   DCFG_ARM_EABI_UNWINDER=false
-   DRUNTIME_OBJS="$DRUNTIME_OBJS gcc/unwind/generic.o"],
-  [AC_MSG_RESULT([yes])
-   DCFG_ARM_EABI_UNWINDER=true
-   DRUNTIME_OBJS="$DRUNTIME_OBJS gcc/unwind/arm.o"])
-AC_SUBST(DCFG_ARM_EABI_UNWINDER)
-
-case "$d_target_os" in
-  aix*|*bsd*|cygwin*|darwin*|gnu*|linux*|skyos*|*solaris*|sysv*) d_have_unix=1 ;;
-esac
-
-if test -n "$d_have_unix" && test "$enable_unix" = auto ; then
-  enable_unix=yes
-fi
-
-case "$d_target_os" in
-    aix*)    d_is_aix=1
-             ;;
-    darwin*) DRUNTIME_OBJS="$DRUNTIME_OBJS \$(RT_OSX_OBJS)"
-             D_EXTRA_OBJS="$D_EXTRA_OBJS \$(OSX_OBJS)"
-             ;;
-    freebsd*|k*bsd*-gnu)
-             D_EXTRA_OBJS="$D_EXTRA_OBJS \$(FREEBSD_OBJS)"
-             ;;
-    linux*)  D_EXTRA_OBJS="$D_EXTRA_OBJS \$(LINUX_OBJS)"
-             ;;
-    mingw*)  DRUNTIME_OBJS="$DRUNTIME_OBJS \$(RT_STDC_OBJS)"
-             DRUNTIME_OBJS="$DRUNTIME_OBJS \$(RT_WINDOWS_OBJS)"
-             D_EXTRA_OBJS="$D_EXTRA_OBJS \$(WINDOWS_OBJS)"
-             ;;
-    *solaris*)
-	    D_EXTRA_OBJS="$D_EXTRA_OBJS \$(RT_SOLARIS_OBJS)"
-esac
-
-if test "$enable_unix" = "yes"; then
-  DRUNTIME_OBJS="$DRUNTIME_OBJS \$(RT_STDC_OBJS)"
-  DRUNTIME_OBJS="$DRUNTIME_OBJS \$(RT_POSIX_OBJS)"
-fi
-
+# Enable libtool
+LT_INIT
+AM_PROG_LIBTOOL
+WITH_LOCAL_DRUNTIME([LT_LANG([D])], [])
+AM_CONDITIONAL([ENABLE_SHARED], [test "$enable_shared" = "fooyes"])
```

Comment:
This should be always false for now. We don't want to build the shared libraries, even if the users uses --enable-shared. I'll change it to `AM_CONDITIONAL([ENABLE_SHARED], [false])`, that's more self-explaining.